sagemaker-python-sdk/mxnet_gluon_embedding_server/mxnet_embedding_server.ipynb

```diff
@@ -0,0 +1,528 @@
+    "\n",
+    "\n",
+    "def model_fn(model_dir):\n",
+    "    \"\"\"\n",
+    "    Load the gluon model. Called once when hosting service starts.\n",
+    "    :param: model_dir The directory where model files are stored.\n",
+    "    :return: a model (in this case a Gluon network)\n",
+    "    \n",
+    "    lil hack: we pass gluon model name as artifact name so that at serving\n",
+    "    we know which model to use\n",
+    "    \"\"\"\n",
+    "    filename = os.listdir(model_dir)[0]\n",
+    "    print('file found in model_dir: ' + filename)\n",
+    "    modelname = filename[:filename.find('.params')]\n",
+    "    net = models.get_model(name=modelname, pretrained=False, ctx=mx.cpu())\n",
+    "    net.load_parameters(os.path.join(model_dir, filename))\n",
+    "    logging.info('loaded parameters into model ' + filename)\n",
+    "\n",
+    "    return net\n",
+    "\n",
+    "\n",
+    "def input_fn(request_body, request_content_type):\n",
+    "    \"\"\"prepares the input\"\"\"\n",
+    "    output = mx.image.imdecode(request_body)\n",
+    "    logging.info('input_fn returns NDArray of shape ' + str(output.shape))\n",
+    "    return output\n",
+    "\n",
+    "\n",
+    "def predict_fn(input_object, model):\n",
+    "    \"\"\"function used for prediction\"\"\"\n",
+    "    emb = model.features\n",
+    "    return embeddings(emb, input_object)\n",
+    "\n",
+    "\n",
+    "\n",
+    "\n",
+    "# Training (no training, just copy from zoo to S3) * * * * * * * * * * * * * * * *\n",
+    "\n",
+    "if __name__ == '__main__':\n",
+    "\n",
+    "    parser = argparse.ArgumentParser()\n",
+    "\n",
+    "    parser.add_argument('--gluon-model-name', type=str, default='resnet18_v2')\n",
+    "    parser.add_argument('--model-dir', type=str, default=os.environ['SM_MODEL_DIR'])\n",
+    "\n",
+    "    args, _ = parser.parse_known_args()\n",
+    "\n",
+    "\n",
+    "    # extract model\n",
+    "    net = models.get_model(name=args.gluon_model_name, pretrained=True, ctx=mx.cpu())\n",
+    "\n",
+    "    # save model\n",
+    "    net.save_parameters(os.path.join(args.model_dir, args.gluon_model_name + '.params'))"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Instantiate a SageMaker Estimator, train and deploy\n",
+    "Training is not actually training a model, it is just copying a model from gluon model zoo to S3"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import sagemaker\n",
+    "from sagemaker import get_execution_role\n",
+    "from sagemaker.mxnet.model import MXNetModel\n",
+    "from sagemaker.mxnet.estimator import MXNet\n",
+    "\n",
+    "sess = sagemaker.Session()\n",
+    "\n",
+    "embedding_model = MXNet(\n",
+    "    entry_point='embedding_server.py',\n",
+    "    py_version='py3',\n",
+    "    framework_version='1.3',\n",
```

Comment:
can we use 1.4 which is the latest version?

Reply:
Hi @yuanzhua , thanks for following-up! I actually did a bit more than updating version:

1. I replaced the training API copy part by a tar compression and copy to S3. Faster and simpler.
2. I enriched the documentation
3. I switched to MXNet 1.4

Ran successfully end-to-end in eu-west-1, us-east-1, ap-southeast-2
let me know next steps
cheers